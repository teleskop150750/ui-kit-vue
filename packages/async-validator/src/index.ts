export * from './types'
export * from './useAsyncValidator'
