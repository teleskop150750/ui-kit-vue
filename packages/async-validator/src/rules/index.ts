import { enumerable } from './enum'
import { pattern } from './pattern'
import { range } from './range'
import { required } from './required'
import { type } from './type'
import { whitespace } from './whitespace'

export const rules = {
  required,
  whitespace,
  type,
  range,
  enum: enumerable,
  pattern,
}
