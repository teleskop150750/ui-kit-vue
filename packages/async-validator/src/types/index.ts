export * from './message'
export * from './rule'
export * from './typescript'
export * from './validate'
export * from './values'
