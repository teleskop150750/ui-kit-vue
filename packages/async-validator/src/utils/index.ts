export * from './error'
export * from './is-type'
export * from './merge'
export * from './message'
export * from './warning'
