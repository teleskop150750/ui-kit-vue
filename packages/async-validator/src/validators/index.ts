import type { ExecuteValidator } from '../types'
import { any } from './any'
import { array } from './array'
import { boolean } from './boolean'
import { date } from './date'
import { enumerable } from './enum'
import { float } from './float'
import { integer } from './integer'
import { method } from './method'
import { number } from './number'
import { object } from './object'
import { pattern } from './pattern'
import { regexp } from './regexp'
import { required } from './required'
import { string } from './string'
import { type } from './type'

type DefaultValidators =
  | 'string'
  | 'method'
  | 'number'
  | 'boolean'
  | 'regexp'
  | 'integer'
  | 'float'
  | 'array'
  | 'object'
  | 'enum'
  | 'pattern'
  | 'date'
  | 'url'
  | 'hex'
  | 'email'
  | 'required'
  | 'any'

type HintedString<KnownValues extends string> = (string & {}) | KnownValues

export const validators: Record<HintedString<DefaultValidators>, ExecuteValidator> = {
  string,
  method,
  number,
  boolean,
  regexp,
  integer,
  float,
  array,
  object,
  enum: enumerable,
  pattern,
  date,
  url: type,
  hex: type,
  email: type,
  required,
  any,
}
