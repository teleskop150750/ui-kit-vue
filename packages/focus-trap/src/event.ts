import type { FocusableElement } from '@nado/tabbable'

export function isEscapeEvent(evt: KeyboardEvent) {
  return evt.key === 'Escape' || evt.key === 'Esc' || evt.keyCode === 27
}

export function isTabEvent(evt: KeyboardEvent) {
  return evt.key === 'Tab' || evt.keyCode === 9
}

// checks for TAB by default
export function isKeyForward(evt: KeyboardEvent) {
  return isTabEvent(evt) && !evt.shiftKey
}

// checks for SHIFT+TAB by default
export function isKeyBackward(evt: KeyboardEvent) {
  return isTabEvent(evt) && evt.shiftKey
}

export function getActualTarget(event: Event) {
  // NOTE: If the trap is _inside_ a shadow DOM, event.target will always be the
  //  shadow host. However, event.target.composedPath() will be an array of
  //  nodes "clicked" from inner-most (the actual element inside the shadow) to
  //  outer-most (the host HTML document). If we have access to composedPath(),
  //  then use its first element; otherwise, fall back to event.target (and
  //  this only works for an _open_ shadow DOM; otherwise,
  //  composedPath()[0] === event.target always).
  const response =
    (event.target as HTMLElement).shadowRoot && typeof event.composedPath === 'function'
      ? event.composedPath()[0]
      : event.target!

  return response as FocusableElement
}
