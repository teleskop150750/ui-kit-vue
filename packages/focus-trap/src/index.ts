export * from './event'
export * from './focus-trap'
export * from './node'
export * from './types'
