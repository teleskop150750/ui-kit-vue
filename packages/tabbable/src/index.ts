// @see https://github.com/focus-trap/tabbable
export * from './candidates'
export * from './constants'
export * from './dom'
export * from './match'
export * from './node'
export * from './sort'
export * from './tabbable'
export * from './types'
