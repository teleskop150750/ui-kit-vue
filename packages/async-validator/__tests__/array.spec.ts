/* eslint-disable @typescript-eslint/no-explicit-any */
import { describe, expect, it } from 'vitest'

import { useAsyncValidator } from '../src'

describe('array', () => {
  it('works for type', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            type: 'array',
          },
        })
        .validate(
          {
            v: '',
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res.length).toBe(1)
    expect(res[0].message).toBe('v is not an array')
  })

  it('works for type and required', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            required: true,
            type: 'array',
          },
        })
        .validate(
          {
            v: '',
          },
          (errors, fields) => {
            res = { errors, fields }
          },
        )
    } catch {}
    expect(res.errors.length).toBe(1)
    expect(res.fields).toMatchInlineSnapshot(`
      Object {
        "v": Array [
          Object {
            "field": "v",
            "fieldValue": "",
            "message": "v is not an array",
          },
        ],
      }
    `)
    expect(res.errors[0].message).toBe('v is not an array')
  })

  it('works for none require', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            type: 'array',
          },
        })
        .validate(
          {
            v: [],
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res).toBe(undefined)
  })

  it('works for empty array', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            required: true,
            type: 'array',
          },
        })
        .validate(
          {
            v: [],
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res.length).toBe(1)
    expect(res[0].message).toBe('v is required')
  })

  it('works for undefined array', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            type: 'array',
          },
        })
        .validate(
          {
            v: undefined,
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res).toBe(undefined)
  })

  it('works for undefined array and required', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            required: true,
            type: 'array',
          },
        })
        .validate(
          {
            v: undefined,
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res.length).toBe(1)
    expect(res[0].message).toBe('v is required')
  })

  it('works for undefined array and defaultField', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            type: 'array',
            defaultField: { type: 'string' },
          },
        })
        .validate(
          {
            v: undefined,
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res).toBe(undefined)
  })

  it('works for undefined array', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            required: true,
            type: 'array',
          },
        })
        .validate(
          {
            v: undefined,
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res.length).toBe(1)
    expect(res[0].message).toBe('v is required')
  })

  it('works for none empty', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            required: true,
            type: 'array',
            message: 'haha',
          },
        })
        .validate(
          {
            v: [1],
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res).toBe(undefined)
  })

  it('works for empty array with min', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            min: 1,
            max: 3,
            type: 'array',
          },
        })
        .validate(
          {
            v: [],
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res.length).toBe(1)
    expect(res[0].message).toBe('v must be between 1 and 3 in length')
  })

  it('works for empty array with max', async () => {
    let res: any

    const validate = useAsyncValidator()

    try {
      await validate
        .useSchema({
          v: {
            min: 1,
            max: 3,
            type: 'array',
          },
        })
        .validate(
          {
            v: [1, 2, 3, 4],
          },
          (errors) => {
            res = errors
          },
        )
    } catch {}
    expect(res.length).toBe(1)
    expect(res[0].message).toBe('v must be between 1 and 3 in length')
  })
})
