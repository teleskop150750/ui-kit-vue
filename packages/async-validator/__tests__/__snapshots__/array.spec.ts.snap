// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`array > works for type and required 1`] = `
{
  "v": [
    {
      "field": "v",
      "fieldValue": "",
      "message": "v is not an array",
    },
  ],
}
`;
